maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/booterclient/ForkStarter.java

```diff
@@ -609,7 +605,7 @@ private RunResult fork( Object testSet, KeyValueSource providerProperties, ForkC
                     new NativeStdErrStreamConsumer( forkClient.getDefaultReporterFactory() );
 
             CommandLineCallable future =
-                    executeCommandLineAsCallable( cli, testProvidingInputStream, threadedStreamConsumer,
+                    executeCommandLineAsCallable( cli, (InputStream) commandSender, threadedStreamConsumer,
```

Comment:
This explicit cast is suspect

Reply:
this is only temporal because the abstraction is not well used in this class.